net/rpc/server.go

```diff
@@ -191,13 +191,34 @@ type Server struct {
 	freeReq    *Request
 	respLock   sync.Mutex // protects freeResp
 	freeResp   *Response
+
+	serverServiceCallInterceptor ServerServiceCallInterceptor
 }
 
 // NewServer returns a new Server.
 func NewServer() *Server {
-	return &Server{}
+	return NewServerWithOpts()
+}
+
+// NewServerWithOpts returns a new Server with the following functional options.
```

Comment:
I'm a bit confused about this extra layer of abstraction, do we intend to add other kind of options (other then interceptors)? if not, why not make an interceptor interface that define `serverServiceCallInterceptor` have a `NewServerWithInterceptors` that take that interface and remove the extra wrapper layer `WithServerServiceCallInterceptor`.
I understand we are trying to make something similar to gRPC here, but I think the reason [GRPC](https://pkg.go.dev/google.golang.org/grpc) went with this design is that they are trying to be as generic as possible. What is the reasoning of going with the same design here?

Reply:
We were chatting about this here:

* https://github.com/hashicorp/consul-net-rpc/pull/8#discussion_r814329002

tl;dr: we do plan on adding more options, possibly. If not, we can revisit this in an upcoming PR.